common/src/main/java/org/jboss/hal/testsuite/cli/StandaloneManager.java

```diff
@@ -128,4 +127,41 @@ public boolean waitUntilAvailable(long timeout) {
         }
         return isRunning;
     }
+
+    @Override
+    public void reloadIfRequiredAndWaitUntilRunning(long timeout) {
+        if (isReloadRequired()) {
+            reloadAndWaitUntilRunning(timeout);
+        }
+    }
+
+    @Override
+    public void reloadAndWaitUntilRunning(long timeout) {
+        reloadServer();
+        waitUntilRunning(timeout);
+    }
+
+    /**
+     * Waits until server is in running state
+     * @throws TimeoutException
+     */
+    public void waitUntilRunning(final long timeout) throws TimeoutException {
+        long decTimeout = timeout;
+        final int step = 500;
+        while (!isInRunningState()) {
+            if (decTimeout <= 0) {
+                throw new TimeoutException("Waiting for running server timed out!");
+            }
+            decTimeout -= step;
+            Library.letsSleep(step);
+        }
+    }
+
+    private boolean isReloadRequired() {
+        return cliClient.reloadRequired();
+    }
+
+    public void reloadServer() {
+        cliClient.reload(true);
+    }
```

Comment:
One more thing, I overlooked, the implementation in `StandaloneManager` just mimics CliClient.reload methods. as such they are not really needed.

E.g. `waitUntilRunning` is more or less the same as `waitUntilAvailable`